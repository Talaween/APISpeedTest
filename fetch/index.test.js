const fetch = require('jest-fetch-mock');

jest.setMock('node-fetch', fetch);

const api = require('./index');

describe('Fetching One URL Data', () => {
 
  test('fetchOne()', async () => {
    
    data = {
            "title": "Yahoo",
            "response_code": 200,
            "speed_score": 69,
            "url": "https://www.yahoo.com/"
        }
    
    fetch.mockResponse(JSON.stringify(data));

    const current = await api.fetchOne('http://yahoo.com');

    expect(current).toEqual(data);
    expect(current).toMatchSnapshot();
  });

  test('fetch', () => {
    expect(fetch).toHaveBeenCalled();
    expect(fetch.mock.calls).toMatchSnapshot();
  });
});