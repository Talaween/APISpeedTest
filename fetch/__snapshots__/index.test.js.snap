// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetching Data fetch 1`] = `
Array [
  Array [
    "https://www.googleapis.com/pagespeedonline/v4/runPagespeed?url=http://yahoo.com",
    Object {
      "method": "GET",
    },
  ],
]
`;

exports[`Fetching Data fetchOne() 1`] = `
Object {
  "response_code": 200,
  "speed_score": 69,
  "title": "Yahoo",
  "url": "https://www.yahoo.com/",
}
`;
